tests/python/op/test_nn.py

```diff
@@ -100,6 +100,22 @@ def forward(self, x_input):
     x = torch.randint(10, (3, 3))
     verify_step(Model(), [x], jit_script=False)
 
+@pytest.mark.parametrize("shape", [(3, 3, 3, 3)])
+def test_matmul(shape):
+    class Model(nn.Module):
+        def __init__(self):
+            super().__init__()
+
+        def forward(self, x_input, y_input):
+            return torch.matmul(x_input, y_input)
+
+    for i in range(1, len(shape) + 1):
+        x_s = shape[::i]
+        x = torch.randn(x_s)
+        for j in range(1, len(shape) + 1):
+            y_s = shape[::j]
+            y = torch.randn(y_s)
+            verify_step(Model(), [x, y], jit_script=False)
```

Comment:
IIUC, you are trying to cover all cases in different n-dims in this loop. However, this is hard to understand and debug in the future. It would be better to explicitly assign shapes in `pytest.mark.parameterize`.

Reply:
Sure I can adjust, but fyi the pytest will have a very long list of shapes. Unless you are thinking of a different approach or there is some macro, we will need to write like ("x_shape" [(3,), (3,), (3,), (3,), (3,3)...]) ("y_shape", [(3,), (3,3)...]) in other words 16 shapes.

Reply:
I didn't see the problem with the long list of shapes?
Talking about the total number of tests that make slow down the CI, your current implementation actually has the same number of workloads.
Talking about the format, `black` should auto-format it to multiple lines.

Reply:
Okay sounds good, I just wanted to confirm if there were any macros or more elegant to write the test case.